sonolark/lib/kube/kube.go

```diff
@@ -731,3 +741,80 @@ func (m *kubePackage) kubeGet(ctx context.Context, r *apiResource, wait time.Dur
 
 	// not reachable
 }
+
+func (m *kubePackage) kubePortForwardTestFn(t *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var namespace, pod, testPath string
+	var localPort, podPort int
+	unpacked := []interface{}{
+		"namespace", &namespace,
+		"pod", &pod,
+		"local_port", &localPort,
+		"pod_port", &podPort,
+		"test_path", &testPath,
+	}
+	if err := starlark.UnpackArgs(b.Name(), args, kwargs, unpacked...); err != nil {
+		return starlark.False, fmt.Errorf("<%v>: %v", b.Name(), err)
+	}
+
+	// check if test path is already reachable before port forward
+	testUrl := fmt.Sprintf("http://localhost:%v%v", localPort, testPath)
+
+	req, err := http.NewRequest(http.MethodGet, testUrl, nil)
+	if err != nil {
+		return starlark.False, err
+	}
+
+	ctx := t.Local(addon.GoCtxKey).(context.Context)
+	_, err = m.httpClient.Do(req.WithContext(ctx))
+	if err == nil {
+		return starlark.False, fmt.Errorf("endpoint %v already in use", testUrl)
+	}
+
+	// port forward the pod
+	roundTripper, upgrader, err := spdy.RoundTripperFor(m.config)
+	if err != nil {
+		return starlark.False, err
+	}
+
+	portForwardPath := fmt.Sprintf("/api/v1/namespaces/%v/pods/%v/portforward", namespace, pod)
+
+	host := strings.Replace(m.Master, "https://", "", 1)
```

Comment:
nit: strings.TrimPrefix would be the function you're looking for here. Either works in this context, but wanted you to be aware of the other method.